### PR TITLE
Page about Bio.Alphabet removal

### PR DESCRIPTION
**Pull request by peterjc**
Intended to replace https://github.com/biopython/biopython/issues/3156 as a place to direct people on code changes from the removal of `Bio.Alphabet` in Biopython 1.78

### CODE REVIEW

File: wiki/Alphabet.md [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+from Bio import SeqIO
+# This file has a single record only
+record = SeqIO.read("Tests/Fasta/wisteria.nu", "fasta")
+rec_start = record[:20]
+rec_start.annotations["molecule_type"] = "DNA"
+SeqIO.write(rec_start, "start_only.xml", "seqxml")
+```
+
+Similarly, the ``Bio.SeqIO.convert`` function's optional alphabet argument has
+been  replaced with an optional molecule type argument:
```
Comment (MarkusPiotrowski) [inline wiki/Alphabet.md:153]: double space

File: wiki/Alphabet.md [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+cope with the removal of the ``Bio.Alphabet`` module in Biopython 1.78
+(September 2020).
+
+The objects from `Bio.Alphabet` had two main uses:
+* Record the molecule type (DNA, RNA or protein) of a sequence,
+* Declare the expected characters in a sequence, alignment, motif, etc.
+
+# Motivation
+
+There were drawbacks to the twenty year old design. In particular how the
```
Comment (mdehoon) [inline wiki/Alphabet.md:21]: I think the bigger issue is that the purpose of alphabets was not defined. Sure, the implementation was also problematic, but the implementation could have been fixed if we had been able to figure out what alphabets were supposed to do in the first place.

File: wiki/Alphabet.md [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+`AlphabetEncoder` class (used to add information about gap or stop symbols)
+was overly complicated, and made even determining the molecule type difficult.
+Taking the consensus of multiple alphabet objects (for example during string
+addition) was also complicated. While you could specify a strict alphabet like
+unambiguous IUPC DNA for a sequence, this did not enforce that only letters
+used were `A`, `C`, `G` and `T`.
+
+# Code changes
+
+After failing to reach any consensus on how to improve or replace the existing
```
Comment (mdehoon) [inline wiki/Alphabet.md:31]: I don't think a lack of consensus was a major issue. The decision to remove alphabets was made after we called for a Biopython Enhancement Proposal to define unambiguously the purpose of alphabets. One person volunteered, and that person in the end came to the conclusion that alphabets were ill-advised. Based on that, and on the fact that nobody else stepped up to define alphabets, we decided to remove them.

File: wiki/Alphabet.md [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+```python
+# New style
+from Bio.Seq import Seq
+my_dna = Seq("ACGT=TT")
+print(my_dna.ungap("="))
+```
+
+## SeqRecord changes
+
+Some sequence file formats require the molecule type, which previously was
```
Comment (mdehoon) [inline wiki/Alphabet.md:81]: Some sequence file formats require *when writing a file*

**Review by mdehoon — COMMENTED** [review]
Left some comments in the text

File: wiki/Alphabet.md (hunk 3 above)
Comment (peterjc) [reply wiki/Alphabet.md:31]: Would you prefer this wording:

"With no clear proposal on how to improve or replace the existing system, it was agreed to remove it."

File: wiki/Alphabet.md (hunk 2 above)
Comment (peterjc) [reply wiki/Alphabet.md:21]: Precede this with "The intended purpose of the alphabet objects was never clearly defined, and ..."?

**Comment by peterjc** [comment]
Suggestions proposed as follow up comments.

File: wiki/Alphabet.md (hunk 2 above)
Comment (mdehoon) [reply wiki/Alphabet.md:21]: ok

File: wiki/Alphabet.md (hunk 3 above)
Comment (mdehoon) [reply wiki/Alphabet.md:31]: ok

**Comment by peterjc** [comment]
Changes made - OK to squash-and-merge now?

**Comment by mdehoon** [comment]
ok

**Comment by peterjc** [comment]
Thank you,

**Pull request merged**